os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -114,14 +114,14 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
 		mm_takesemaphore(heap);
 
 		prev = NULL;
-		node = heap->mm_heapstart[region];
+		node = (struct mm_freenode_s *)heap->mm_heapstart[region];
 		next = (struct mm_freenode_s *)((char *)node + node->size);
 
-		for (; node < heap->mm_heapend[region]; prev = node, node = next, next = (struct mm_freenode_s *)((char *)next + next->size)) {
+		for (; node < (struct mm_freenode_s *)heap->mm_heapend[region]; prev = node, node = next, next = (struct mm_freenode_s *)((char *)next + next->size)) {
 			mvdbg("region=%d node=%p size=%p preceding=%p (%c)\n", region, node, node->size, (node->preceding & ~MM_ALLOC_BIT), (node->preceding & MM_ALLOC_BIT) ? 'A' : 'F');
 
-			if ((prev && prev->size != node->preceding & ~MM_ALLOC_BIT) ||
-				(node->size != next->preceding & ~MM_ALLOC_BIT) ||
+			if ((prev && prev->size != (node->preceding & ~MM_ALLOC_BIT)) ||
+				(node->size != (next->preceding & ~MM_ALLOC_BIT)) ||
```

Comment:
Regardless of this PR, how about using macro for preceding size and alloc bit?
```c
#define SIZE_PREV_NODE(x)  (x)->preceding & ~MM_ALLOC_BIT
#define IS_ALLOC_NODE(x)    (x)->preceding & MM_ALLOC_BIT
#define NEXT_NODE(x)          (char *)(x) + (x)->size
```

Reply:
I agree. I will change with another PR. Thanks.